mindsdb/interfaces/agents/callback_handlers.py

```diff
@@ -113,10 +113,18 @@ def on_agent_action(self, action: AgentAction, **kwargs: Any) -> Any:
         '''Run on agent action.'''
         self.logger.debug(f'Running tool {action.tool} with input:')
         self.logger.debug(action.tool_input)
+
+        stop_block = 'Observation: '
```

Comment:
@dusvyat, I'm not sure about this. Maybe there is a better place to locate stop word

Reply:
https://python.langchain.com/v0.1/docs/modules/callbacks/

perhaos we can use:

```
    def on_agent_finish(self, finish: AgentFinish, **kwargs: Any) -> Any:
        """Run on agent end."""


```

Reply:
on agent finish - is the end of agent's work
but I wan't to modify action input, remove extra data from it (problem 6 in pr description)